src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -417,57 +482,26 @@ protected Expr createTransportSettingsInitExpr(
 
-    // Generate changes opName
-    // This will be replaced and edited based on annotations
-    MethodInvocationExpr requestGetProjectExpr =
-        MethodInvocationExpr.builder()
-            .setMethodName("getProject")
-            .setExprReferenceExpr(requestVarExpr)
-            .build();
-    ValueExpr colonValueExpr =
-        ValueExpr.builder().setValue(StringObjectValue.builder().setValue(":").build()).build();
-    MethodInvocationExpr opNameAppendColonProjectExpr =
-        MethodInvocationExpr.builder()
-            .setMethodName("append")
-            .setArguments(colonValueExpr)
-            .setExprReferenceExpr(opNameVarExpr)
-            .build();
-    opNameAppendColonProjectExpr =
-        methodMaker
-            .apply("append", Collections.singletonList(requestGetProjectExpr))
-            .apply(opNameAppendColonProjectExpr);
-    createBody.add(ExprStatement.withExpr(opNameAppendColonProjectExpr));
-
-    // Generate changes to opName
-    MethodInvocationExpr requestGetRegionExpr =
-        MethodInvocationExpr.builder()
-            .setMethodName("getRegion")
-            .setExprReferenceExpr(requestVarExpr)
-            .build();
-    MethodInvocationExpr opNameAppendColonRegionExpr =
-        MethodInvocationExpr.builder()
-            .setMethodName("append")
-            .setArguments(colonValueExpr)
-            .setExprReferenceExpr(opNameVarExpr)
-            .build();
-    opNameAppendColonRegionExpr =
-        methodMaker
-            .apply("append", Collections.singletonList(requestGetRegionExpr))
-            .apply(opNameAppendColonRegionExpr);
-    createBody.add(ExprStatement.withExpr(opNameAppendColonRegionExpr));
+    // Generate compound operation name
+    if (!protoMethod.isOperationPollingMethod()) {
+      // TODO: Change to ordered map
+      Map<String, String> requestFields = inputOperationMessage.operationRequestFields();
+      List<String> fieldAnnotationNames = new ArrayList<String>(requestFields.keySet());
+      Collections.sort(fieldAnnotationNames);
+      for (String fieldName : fieldAnnotationNames) {
+        createBody.add(appendField(opNameVarExpr, requestVarExpr, requestFields.get(fieldName)));
+      }
+    }
 
-    // Generate check status expression
-    // This will be replaced and edited based on annotations
+    // Generate check for status == done
     MethodInvocationExpr getStatusExpr =
-        MethodInvocationExpr.builder()
-            .setExprReferenceExpr(responseVarExpr)
-            .setMethodName("getStatus")
-            .build();
+        invokeMethodfromVar(
+            responseVarExpr, getMethodFormat(operationResponse.getStatusFieldName()));
     TypeNode statusType =
         TypeNode.withReference(
             VaporReference.builder()
                 .setName("Status")
-                .setPakkage("com.google.cloud.compute.v1")
+                .setPakkage("com.google.cloud.compute.v1.Operation")
```

Comment:
All explicit mentions of "compute" must be gone. Why is this still needed?
Please check for all other mentions of compute in your codebase. There should be 0 mentions of compute in gapic-generator-java codebase.

Reply:
So status is generated somewhere else in compute, and is used in this class. I am not using compute to generate anything. This is a Vapor reference and that is just the package that will be imported in the generated class. 

See: https://github.com/vam-google/google-cloud-compute-small-v1-lro/blob/7ff5244a579476088a222c50a7155e47e107a277/gapic-google-cloud-compute-small-v1-java/src/main/java/com/google/cloud/compute/v1/stub/HttpJsonAddressesStub.java#L46